ando/tools/generator/tests/test_utils.py

```diff
@@ -0,0 +1,70 @@
+import unittest
+
+from ando.tools.generator.tests.utils import *
+from ando.tools.generator.utils import *
+import pandas as pd
+import os
+import json
+from pathlib import Path
+import copy
+
+
+class TestUtils(unittest.TestCase):
+
+    def setUp(self) -> None:
+        initialize_test_directory()
+
+    def test_create_file(self):
+        df = pd.DataFrame()
+        path_to_save = os.path.join(test_directory, "test_create_file.tsv")
+        save_tsv(df, path_to_save)
+        self.assertTrue(Path(path_to_save).exists())
+
+    def test_create_dummy_file(self):
+        df = pd.read_csv(os.path.join(test_directory / 'test_files' / 'participants.tsv'), sep='\t')
+        path_to_save = os.path.join(test_directory, "test_create_dummy_file.tsv")
+        save_tsv(df, path_to_save)
+
+        self.assertTrue(Path(path_to_save).exists())
+        df_test = pd.read_csv(os.path.join(test_directory, "test_create_dummy_file.tsv"), sep='\t')
+
+        self.assertEqual(set(df.columns), set(df_test.columns))
+        self.assertEqual(df.shape, df_test.shape)
+        self.assertEqual(set(df), set(df_test))
+
+    def test_create_dummy_file_existing(self):
```

Comment:
This name seems a bit misleading as you are specifying the complete result of the merge operation and not just the fact that the method does not fail if the `dummy` file already exists..

Reply:
updated